src/main/java/org/apache/maven/shared/verifier/Verifier.java

```diff
@@ -1566,38 +1567,76 @@ public void reset()
         }
     }
 
+    /**
+     * @deprecated will be removed, please use {@link #addCliArgument(String)}, {@link #addCliArguments(String...)}
+     */
+    @Deprecated
     public List<String> getCliOptions()
```

Comment:
I think the comment is partially wrong. You describe one usecase, but the other one is really to get the args again, for read-only purposes. That should be reflected. For now, we don't need a replacement. (read-only getter)

Reply:
I want to remove this methods at all - so there will no be replacement for it. 

I don't see a use case for read  only arguments list.

Reply:
Agree here

Reply:
Then the comment should say to be removed without replacement

Reply:
updated